### ISSUE TITLE
PackAsTool fails with Microsoft.EntityFrameworkCore dependency

### ISSUE THREAD

**Issue by tthiery** [dotnet/efcore#19576]
When trying to pack a global dotnet tool with a dependency on `Microsoft.EntityFrameworkCore` the package process fails. This happens only with the dependency but not without. Other dependencies work fine.

### Steps to reproduce
1. Have 3.1.100 installed
1. Execute the following code

```sh
dotnet new console
dotnet add package Microsoft.EntityFrameworkCore
dotnet pack /p:PackAsTool=true
```
The last command fails with the following message
````
C:\Program Files\dotnet\sdk\3.1.100\Sdks\NuGet.Build.Tasks.Pack\build\NuGet.Build.Tasks.Pack.targets(198,5): error : The DateTimeOffset specified cannot be converted into a Zip file timestamp. (Parameter 'value') [C:\dev\tbd\foo\foo.csproj]
````

(for a positive outcome change the dependency to `McMaster.Extensions.CommandLineUtils`)

### Further technical details

EF Core version: 3.1.100
Database provider: used originally Microsoft.EntityFrameworkCore.SqlServer but fails also without.
Target framework: .NET Core 3.1
Operating system: Windows
IDE: Visual Studio Code / CLI

dotnet --list-sdks
2.1.700 [C:\Program Files\dotnet\sdk]
2.2.104 [C:\Program Files\dotnet\sdk]
2.2.300 [C:\Program Files\dotnet\sdk]
3.1.100 [C:\Program Files\dotnet\sdk]

**Comment by ajcvickers** [comment]
Duplicate of #19542

**Comment by tthiery** [comment]
Resolved in version 3.1.1
